cpp/src/io/parquet/parquet.hpp

```diff
@@ -261,6 +261,67 @@ struct Statistics {
   thrust::optional<std::vector<uint8_t>> min_value;
 };
 
+/**
+ * @brief Thrift-derived struct containing statistics used to estimate page and column chunk sizes
+ */
+struct SizeStatistics {
+  // Number of variable-width bytes stored for the page/chunk. Should not be set for anything
+  // but the BYTE_ARRAY physical type.
+  thrust::optional<int64_t> unencoded_byte_array_data_bytes;
+  /**
+   * When present, there is expected to be one element corresponding to each
+   * repetition (i.e. size=max repetition_level+1) where each element
+   * represents the number of times the repetition level was observed in the
+   * data.
+   *
+   * This value should not be written if max_repetition_level is 0.
+   */
+  thrust::optional<std::vector<int64_t>> repetition_level_histogram;
```

Comment:
Is there a reason to mix `thrust::` with `std::`? Why don't just use `std::optional`?

Reply:
The compact protocol reader/writer uses `thrust::optional` because some optional fields need to be usable on the device.

Reply:
But how `thrust::optional<std::vector>>` can be used on device? Or this is just for consistency with other `thrust::optional<plain_type>`?

Reply:
Yes, consistency. LogicalType, for instance, is optional as well, and is referenced in device code. Since that one had to be `thrust::optional`, I figured it was easier for them all to be.